### PR TITLE
Discard down message with a different ref

### PR DESCRIPTION
**Pull request by samphilipd**

### CODE REVIEW

**Comment by CrowdHailer** [comment]
have you tried running with this branch to see if it does solve the problem?

If at all possible I'd like a test or possibly test an indication that it does work.

**Comment by CrowdHailer** [comment]
Also thanks btw :-)

If no tests obvious lets just add a comment in the clause quickly saying why it was added

**Comment by CrowdHailer** [comment]
Also can you rebase against 0.9.x? you branched off a slightly stale version, around 0.9.1 I think

**Comment by samphilipd** [comment]
It is rebased off latest 0.9.x.

I have absolutely no idea how to test it, I don't even really know why it's happening in the first place. Seems to be random.

I will try running with this branch now.

**Comment by CrowdHailer** [comment]
Check the network diagram there are a few more commits on 0.9x. perhaps your fork is out of date. 
https://github.com/CrowdHailer/Ace/network

**Comment by CrowdHailer** [comment]
@samphilipd did you force push after the rebase? turns out I get no notification of you doing that from gihub

**Comment by CrowdHailer** [comment]
looks good, can you bump the patch version and add an entry to the changelog.

**Comment by samphilipd** [comment]
@CrowdHailer Yes I did, and done.

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,11 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](http://keepachangelog.com/)
 and this project adheres to [Semantic Versioning](http://semver.org/).
 
+## [0.9.3](https://github.com/CrowdHailer/Ace/tree/0.9.1) - 2017-06-14
```
Comment (CrowdHailer) [inline CHANGELOG.md]: needs to link to 0.9.3

Comment (samphilipd) [inline CHANGELOG.md]: doh

**Comment by CrowdHailer** [comment]
thanks :100:

**Pull request merged**